### PR TITLE
Fix/tfd 147 add kerberos support

### PR DESCRIPTION
**Pull request by RyanSkraba**
**Please check if the PR fulfills these requirements**

- [X] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

Add kerberos support to TCOMP HDFS components.

**Does this PR introduce a breaking change?**

- [ ] Yes
- [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/554/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/554/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/563/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/563/)

File: components/components-fileio/simplefileio-definition/src/main/java/org/talend/components/simplefileio/SimpleFileIoDatastoreProperties.java [hunk 1]
```diff
@@ -32,6 +38,22 @@ public void setupLayout() {
...
+        mainForm.addRow(kerberosPrincipal);
+        mainForm.addRow(kerberosKeytab);
     }
 
+    @Override
+    public void refreshLayout(Form form) {
+        super.refreshLayout(form);
+        // Main properties
+        if (form.getName().equals(Form.MAIN)) {
+            form.getWidget(kerberosPrincipal.getName()).setVisible(useKerberos.getValue());
```
Comment (pbailly) [inline components/components-fileio/simplefileio-definition/src/main/java/org/talend/components/simplefileio/SimpleFileIoDatastoreProperties.java]: I think you can even do:
`form.getWidget(kerberosPrincipal.getName()).setVisible(useKerberos);`

File: components/components-fileio/simplefileio-definition/src/test/java/org/talend/components/simplefileio/SimpleFileIoDatastorePropertiesTest.java [hunk 2]
```diff
@@ -76,6 +76,7 @@ public void testSetupLayout() {
     @Test
     public void testRefreshLayout() {
         Form main = properties.getForm(Form.MAIN);
+        properties.useKerberos.setValue(false);
```
Comment (pbailly) [inline components/components-fileio/simplefileio-definition/src/test/java/org/talend/components/simplefileio/SimpleFileIoDatastorePropertiesTest.java]: You should test when the visibility is true.

File: components/components-fileio/simplefileio-definition/src/main/java/org/talend/components/simplefileio/SimpleFileIoDatastoreProperties.java [hunk 3]
```diff
@@ -32,6 +38,22 @@ public void setupLayout() {
         super.setupLayout();
         Form mainForm = new Form(this, Form.MAIN);
         mainForm.addRow(userName);
+        mainForm.addRow(useKerberos);
+        mainForm.addRow(kerberosPrincipal);
+        mainForm.addRow(kerberosKeytab);
     }
 
+    @Override
+    public void refreshLayout(Form form) {
```
Comment (pbailly) [inline components/components-fileio/simplefileio-definition/src/main/java/org/talend/components/simplefileio/SimpleFileIoDatastoreProperties.java]: You could hide the username

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/575/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/575/)

**Pull request merged**